src/pynxtools/nomad/parser.py

```diff
@@ -285,6 +309,9 @@ def _populate_data(
                     else:
                         pint_unit = ureg.parse_units("1")
                     field = ureg.Quantity(field, pint_unit)
+                    if field_stats is not None:
+                        for i in range(4):
+                            field_stats[i] = ureg.Quantity(field_stats[i], pint_unit)
```

Comment:
This hardcoding here might break in the future if we change the field values.
Also, var has units of [unit]^2, if [unit] is the unit of the variable (stddev = sqrt(var) would have the same units).

Reply:
addressed